web/src/main/java/it/pagopa/selfcare/party/registry_proxy/web/controller/PDNDController.java

```diff
@@ -0,0 +1,63 @@
+package it.pagopa.selfcare.party.registry_proxy.web.controller;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import it.pagopa.selfcare.party.registry_proxy.connector.model.PDND;
+import it.pagopa.selfcare.party.registry_proxy.connector.model.QueryResult;
+import it.pagopa.selfcare.party.registry_proxy.core.PDNDService;
+import it.pagopa.selfcare.party.registry_proxy.web.model.PDNDsResource;
+import it.pagopa.selfcare.party.registry_proxy.web.model.mapper.PDNDMapper;
+import it.pagopa.selfcare.party.registry_proxy.web.model.mapper.PDNDsMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "/pdnd", produces = MediaType.APPLICATION_JSON_VALUE)
+@Api(tags = "pdnd")
+public class PDNDController {
+
+    private final PDNDService pdndService;
+    private final PDNDMapper pdndMapper;
+
+    @Autowired
+    public PDNDController(PDNDService pdndService,
+                          PDNDMapper pdndMapper) {
+        log.trace("Initializing {}", PDNDController.class.getSimpleName());
+        this.pdndService = pdndService;
+        this.pdndMapper = pdndMapper;
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "${swagger.api.pdnd.search.summary}",
+            notes = "${swagger.api.pdnd.search.notes}")
+    public PDNDsResource search(@ApiParam("${swagger.model.pdnd.search}")
+                                @RequestParam(value = "search", required = false)
+                                Optional<String> search,
+                                @ApiParam(value = "${swagger.model.*.page}")
+                                @RequestParam(value = "page", required = false, defaultValue = "1")
+                                Integer page,
+                                @ApiParam(value = "${swagger.model.*.limit}")
+                                @RequestParam(value = "limit", required = false, defaultValue = "10")
+                                Integer limit) {
+        log.trace("search start");
+        log.debug("search search = {}, page = {}, limit = {}", search, page, limit);
+        final QueryResult<PDND> result = pdndService.search(search, page, limit);
+        final PDNDsResource institutionsResource = PDNDsMapper.toResource(result.getItems().stream()
+                        .map(pdndMapper::toResource)
+                        .collect(Collectors.toList()),
+                result.getTotalHits());
```

Comment:
you might avoid to create a mapper just for setting few data....you could use set method, for example:

pdndResource = new PDNDsResource();
            pdndResource.setItems(result.getItems().stream()
                        .map(pdndMapper::toResource)
                        .collect(Collectors.toList());
            pdndResource.setCount( result.getTotalHits());